.github/workflows/sanitizers.yml

```diff
@@ -0,0 +1,54 @@
+name: Sanitizers
+
+on: [pull_request, push]
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref || github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  build:
+    timeout-minutes: 10
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - sanitizer-name: asan
+            cmake-args: "-DCMAKE_BUILD_TYPE=Asan"
+            optimizer_backend: sleipnir
+            os: ubuntu-24.04
+          - sanitizer-name: tsan
+            cmake-args: "-DCMAKE_BUILD_TYPE=Tsan"
+            optimizer_backend: sleipnir
+            os: ubuntu-24.04
+          - sanitizer-name: ubsan
+            cmake-args: "-DCMAKE_BUILD_TYPE=Ubsan"
+            optimizer_backend: sleipnir
+            os: ubuntu-24.04
+          - sanitizer-name: asan
+            cmake-args: "-DCMAKE_BUILD_TYPE=Asan"
+            optimizer_backend: casadi
+            os: ubuntu-24.04
+          - sanitizer-name: tsan
+            cmake-args: "-DCMAKE_BUILD_TYPE=Tsan"
+            optimizer_backend: casadi
+            os: ubuntu-24.04
+          - sanitizer-name: ubsan
+            cmake-args: "-DCMAKE_BUILD_TYPE=Ubsan"
+            optimizer_backend: casadi
+            os: ubuntu-24.04
```

Comment:
You can just harcode the os in the runs-on line because it'll always be the same for all of them.